core/lib/spree/events/processors/mail_processor.rb

```diff
@@ -0,0 +1,42 @@
+module Spree
+  module Events
+    module Processors
+      class MailProcessor
+        class << self
+          def register!
+            Spree::Config.event_bus.subscribe(:order_confirmed,           ->(event) { send_confirm_email(event) })
+            Spree::Config.event_bus.subscribe(:order_cancelled,           ->(event) { send_cancel_email(event) })
+            Spree::Config.event_bus.subscribe(:carton_shipped,            ->(event) { send_carton_shipped_emails(event) })
+            Spree::Config.event_bus.subscribe(:reimbursement_processed,   ->(event) { send_reimbursement_email(event) })
+            Spree::Config.event_bus.subscribe(:order_inventory_cancelled, ->(event) { send_inventory_cancellation_email(event) })
```

Comment:
is it possible (easy) for stores or extensions to add other subscriptions to this list? One way could be creating a custom `MailProcessor` class that inherits from this one, but we should probably use something like

```ruby
Spree::Config.mail_processor_class.register!
```

instead of

```ruby
Spree::Events::Processors::MailProcessor.register!
```

into `core/engine.rb`

Reply:
I think the idea is to eventually pull this mail processor out entirely, and move it to an solidus_action_mailer extension, eventually, so a Config for mail_processor_class doesn't need to be supported by core at all.

Reply:
The only time you'd need to modify this list specifically was if you wanted to change the `MailProcessor` list, which as @cbrunsdon mentioned eventually might be pulled out entirely. 

However, if all you wanted to do was subscribe to the same event, any other object can call `Spree::Config.event_bus.subscribe(:order_cancelled, OTHER_PROC)` and it will not change the existing subscriptions. You can have any number of subscribers to an event, so a store can fairly easily call that function to add themselves to the list if they had some functionality they wanted to made sure happened in response to the event. The only concern then is that they call `subscribe` early enough in the startup of the app so as to be able to get the events when they happen.

Reply:
I am in favour of removing the class out of core and move this into an action mailer extension instead. Then provide an easy (as in not `class_eval`) configuration for people wanting to enable/disable certain event handlers. But this can be discussed in that new extension instead of core.